File: rb-appscript/tags/rb-appscript-0.1.5/sample/Print_folder_tree.rb
#!/usr/bin/env ruby

# Prints the sub-folder hierarchy of a given folder as a list of folder names indented according to depth.

require "appscript"

def printFolderTree(folder, indent='')
    puts indent + folder.name.get
    folder.folders.get.each { |folder| printFolderTree(folder, indent + "\t") }
end

printFolderTree(AS.app('Finder').home.folders['Documents'])

File: py-appscript/branches/py-appscript-0.16.x/Examples/appscript/Print_folder_tree.py
#!/usr/bin/env python

from appscript import *


def printFolderTree(folder, indent=''):
    """Print a tab-indented list of a folder tree."""
    print indent + folder.name.get().encode('utf8')
    for folder in folder.folders.get():
        printFolderTree(folder, indent + '\t')


printFolderTree(app('Finder').home.folders['Documents'])